### DISCUSSION TITLE
How to do inference with a Flax model trained with HuggingFace?

### DISCUSSION THREAD

**Discussion by ProBuro** [google/flax#1787] [General]
Hello. I somehow trained FLAX BigBird for token classification. I have a saved model with .msgpack extension. All my attempts to infer the model with test data in a separate colab notebook did not lead to anything. Attempts to find at least one example with a detailed description of loading models,
and running a test with a description of all the necessary functions also did not lead to anything. As a result, having spent several days training the model, somehow correcting all the errors that have arisen, now I don’t understand what to do next, having a model and test data. Roughly speaking, it turns out that it makes sense to train FLAX models. Is there somewhere
then an example of pure JAX model output (without training and validation)?

**Comment by marcvanzee** [comment]
>  I somehow trained FLAX BigBird for token classification

Could you be more specific? Did you use HuggingFace, or did you use your own code?

> All my attempts to infer the model with test data in a separate colab notebook did not lead to anything. 

Could you share the Colab?

> Attempts to find at least one example with a detailed description of loading models, and running a test with a description of all the necessary functions also did not lead to anything.

Many of our examples are able to save and load checkpoints and do inference with them, but we of course alway can improve. If you answer the questions above then I might be able to help you better, and perhaps we can improve our documentation so you can find these things easier!

**Reply by ProBuro** [reply]
I trained BigBird HuggingFace by redoing the code from here basis https://github.com/huggingface/transformers/blob/master/examples/flax/token-classification/run_flax_ner.py
and this code was the 
https://www.kaggle.com/asvskartheek/bert-tpus-jax-huggingface
both codes are distinguished by one thing - there is no clean output.
I don't have an output notepad.
when there is a trained model and text files, how to make a clean conclusion with it?
1. upload data
2. create a dataset
3. define the tokenizer
4. define the model
5. How to make STATE for pure test output?
6. How to make TEST_STEP_LOADER? (although I found it, but haven't tested it yet)
7. How to load data into the model?
8. Is it possible to somehow change the input layer to pass additional data through the model

**Reply by marcvanzee** [reply]
Given that you trained your model with HuggingFace, please file an issue against their repo. We don't maintain / support the HuggingFace models / interfaces here, just the Flax code. If you have any Flax-specific questions, feel free to ask!

**Comment by ProBuro** [comment]
so it's by Flax, Jax. It's about there being no guide to clean derivation.
For example: how to create a state for a pure test, then to call
unrep_state = unreplicate(state)
? and other. in the transformers code, only the tokenizer and the model, the rest is all in jax.

**Reply by jheek** [reply]
I don't think this is a helpful discussion. You will have to be more specific in your questions otherwise we cannot provide help.
My advice would be to work through some examples in either flax/examples or huggingface to get a better sense of how to train, finetune, and evaluate models. If there is some specific issues that remains we are happy to help out.

**Reply by ProBuro** [reply]
the question is that the documentation does not address the issue of how the model works with unknown data. not evaluation of the model, namely the operation of the model with unknown data. when the data and the model are taken, the loading of the model and data is determined, and as a result, the result of the work is obtained. validation and training is great, it's there, but what's next? state_train exists - great, but how to use it to process your data after training? without loss_fn, tx ... what does training and validation have to do with it?

**Reply by jheek** [reply]
I think what you are looking for is a guide to fine-tuning. This is definitely something the HuggingFace library and their docs, community can help you with

**Reply by ProBuro** [reply]
ok, thanks.
could you please give me a link to the code that has the Jax, Flax model working with non-training data. Any model that does not work with training data during training (validation), but immediately processes data unknown to it?
